src/components/popups/PopupInlineQuestionInsert.tsx

```diff
@@ -42,23 +46,41 @@ export const PopupInlineQuestionInsert = ({wide, codemirror}: { wide?: boolean,
             <Container className={styles.cmPanelPopup}>
                 <Label for={"inline-part-index"}>Part ID</Label>
                 <Input id={"inline-part-index"} placeholder={"Default"} onChange={(e) => setID(e.target.value)} />
-                <hr/>
-                <Label for={"inline-part-width"}>Width:</Label>
-                <Input id={"inline-part-width"} placeholder={"Default"} onChange={ifValidNumericalInputThen(setWidth)}/>
-                <hr/>
-                <Label for={"inline-part-height"}>Height:</Label>
-                <Input id={"inline-part-height"} placeholder={"Default"} onChange={ifValidNumericalInputThen(setHeight)} />
-                <hr/>
+                <hr className="mb-1"/>
+                {mode === "classes" ? <>
+                    <Label for={"inline-part-classes"}>Classes:</Label>
+                    <Input id={"inline-part-classes"} placeholder={"e.g. wf-8"} onChange={(e) => setClasses(e.target.value)}/>
+                </> : <>
+                    <Label for={"inline-part-width"}>Width (px):</Label>
+                    <Input id={"inline-part-width"} placeholder={"Default (100)"} onChange={ifValidNumericalInputThen(setWidth)}/>
+                    <hr/>
+                    <Label for={"inline-part-height"}>Height (px):</Label>
+                    <Input id={"inline-part-height"} placeholder={"Default (27)"} onChange={ifValidNumericalInputThen(setHeight)}/>
+                </>}
```

Comment:
Do we want to consider some input validation warning text here? e.g. for `classes="wf-21"` or `width="40"`

Obviously the wiki is linked right here with all the info, and this could also be used for custom class input anyway so we wouldn't be covering those cases (so I'm not sure how much help it'd be anyway) - but I'd at least like to know your thoughts?

Reply:
It's a good thought. The intention is to allow any class that exists in the frontend, and given we can't know the set of all classes, as long as the warning is a blacklist of invalid classes it'd be useful. I think we should see if it's something the content team need after letting them try it out; it's hopefully not too hard for them to debug with the wiki page!